operator-common/src/main/java/io/strimzi/operator/common/Operator.java

```diff
@@ -71,16 +72,19 @@ default void reconcileAll(String trigger, String namespace, Handler<AsyncResult<
     default void reconcileThese(String trigger, Set<NamespaceAndName> desiredNames, String namespace, Handler<AsyncResult<Void>> handler) {
         if (desiredNames.size() > 0) {
             List<Future> futures = new ArrayList<>();
-            desiredNames.stream().map(res -> res.getNamespace()).collect(Collectors.toSet()).forEach(ns -> resourceCounter(ns).set(0));
+            desiredNames.stream().map(res -> res.getNamespace()).collect(Collectors.toSet()).forEach(ns -> {
+                resourceCounter(ns).set(0);
+                pausedResourceCounter(ns).set(0);
+            });
 
             for (NamespaceAndName resourceRef : desiredNames) {
-                resourceCounter(resourceRef.getNamespace()).getAndIncrement();
                 Reconciliation reconciliation = new Reconciliation(trigger, kind(), resourceRef.getNamespace(), resourceRef.getName());
                 futures.add(reconcile(reconciliation));
             }
             CompositeFuture.join(futures).map((Void) null).onComplete(handler);
         } else {
             resourceCounter(namespace).set(0);
+            pausedResourceCounter(namespace).set(0);
```

Comment:
How does this work with operator watching all namespaces? Line 76-77 will reset it for namespaces which have some resources. But these two lines would reset it for namespace `*` I suspect.

Reply:
Correct